git-fixup

```diff
@@ -46,116 +55,130 @@ function fixup_candidates_lines () {
 
 # Produce suggestion of commits by taking the latest commit to each file with
 # staged changes
-function fixup_candidates_files () {
+fixup_candidates_files () {
     git diff --cached --name-only | (
-        while read file; do
-            git rev-list $rev_range -- $file \
-                | grep -v -f <(git rev-list -E --grep='^(fixup|squash)' $rev_range -- $file) \
+        TMP_COMMIT_LIST="$(mktemp --tmpdir gitfixup-XXXX)"
```

Comment:
We can probably forget about git 1.8 support at this point and go back to --invert-grep which I hope would avoid this temp file stuff

Ref https://github.com/keis/git-fixup/pull/34/files